pkg/controller/atlasstream/atlasstream_connection_test.go

```diff
@@ -0,0 +1,1093 @@
+				},
+				&http.Response{},
+				nil,
+			)
+		streamsAPI.EXPECT().
+			UpdateStreamConnection(context.Background(), "my-project-id", "instance-0", "sample-connection2", mock.AnythingOfType("*admin.StreamsConnection")).
+			Return(admin.UpdateStreamConnectionApiRequest{ApiService: streamsAPI})
+		streamsAPI.EXPECT().
+			UpdateStreamConnectionExecute(mock.AnythingOfType("admin.UpdateStreamConnectionApiRequest")).
+			Return(
+				&admin.StreamsConnection{
+					Name: pointer.MakePtr("sample-connection2"),
+					Type: pointer.MakePtr("Sample"),
+				},
+				&http.Response{},
+				nil,
+			)
+		streamsAPI.EXPECT().
+			DeleteStreamConnection(context.Background(), "my-project-id", "instance-0", "sample-connection3").
+			Return(admin.DeleteStreamConnectionApiRequest{ApiService: streamsAPI})
+		streamsAPI.EXPECT().
+			DeleteStreamConnectionExecute(mock.AnythingOfType("admin.DeleteStreamConnectionApiRequest")).
+			Return(
+				nil,
+				&http.Response{},
+				nil,
+			)
+		streamsAPI.EXPECT().ListStreamConnections(context.Background(), "my-project-id", "instance-0").
+			Return(admin.ListStreamConnectionsApiRequest{ApiService: streamsAPI})
+		streamsAPI.EXPECT().ListStreamConnectionsExecute(mock.AnythingOfType("admin.ListStreamConnectionsApiRequest")).
+			Return(
+				&admin.PaginatedApiStreamsConnection{
+					Results: &[]admin.StreamsConnection{
+						{
+							Name:        pointer.MakePtr("sample-connection2"),
+							Type:        pointer.MakePtr("Cluster"),
+							ClusterName: pointer.MakePtr("my-cluster"),
+							DbRoleToExecute: &admin.DBRoleToExecute{
+								Role: pointer.MakePtr("readWrite"),
+								Type: pointer.MakePtr("BUILT_IN"),
+							},
+						},
+						{
+							Name: pointer.MakePtr("sample-connection3"),
+							Type: pointer.MakePtr("Sample"),
+						},
+					},
+					TotalCount: pointer.MakePtr(2),
+				},
+				&http.Response{},
+				nil,
+			)
+		ctx := &workflow.Context{
+			Context: context.Background(),
+			SdkClient: &admin.APIClient{
+				StreamsApi: streamsAPI,
+			},
+		}
+		atlasInstance := &admin.StreamsTenant{
+			Name: pointer.MakePtr("instance-0"),
+		}
+
+		result, err := reconciler.handleConnectionRegistry(ctx, project, streamInstance, atlasInstance)
+		assert.NoError(t, err)
+		assert.Equal(t, ctrl.Result{}, result)
+		assert.Equal(t, status.ReadyType, ctx.Conditions()[0].Type)
+		assert.Equal(t, corev1.ConditionTrue, ctx.Conditions()[0].Status)
+		assert.Equal(t, status.StreamInstanceReadyType, ctx.Conditions()[1].Type)
+		assert.Equal(t, corev1.ConditionTrue, ctx.Conditions()[1].Status)
+	})
+
+	t.Run("should transition to terminate state when failing to sort operations", func(t *testing.T) {
+		project := &akov2.AtlasProject{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "my-project",
+				Namespace: "default",
+			},
+			Spec: akov2.AtlasProjectSpec{
+				Name: "my-project",
+			},
+			Status: status.AtlasProjectStatus{
+				ID: "my-project-id",
+			},
+		}
```

Comment:
This tests with pages of setup code are hard to read and many times the setup is repetitive. This project is exactly the same as above right? so why not just this?

```suggestion
		project := defaultTestAtlasProject()
```

Reply:
I'm passing this for now once we are going to discuss this topic externally